src/lib.rs

```diff
@@ -0,0 +1,1175 @@
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true, false, true];
+    ///     v.truncate(2);
+    ///     assert_eq!(v, vob![true, false]);
+    /// }
+    /// ```
+    pub fn truncate(&mut self, len: usize) {
+        if len > self.len {
+            return;
+        }
+        self.len = len;
+        self.vec.truncate(blocks_required::<T>(len));
+        self.mask_last_block();
+    }
+
+    /// Appends a bool to the back of the Vob.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v = Vob::new();
+    /// v.push(true);
+    /// v.push(false);
+    /// assert_eq!(v.get(0), Some(true));
+    /// assert_eq!(v.get(1), Some(false));
+    /// ```
+    pub fn push(&mut self, value: bool) {
+        if self.len % bits_per_block::<T>() == 0 {
+            self.vec.push(T::zero());
+        }
+        let i = self.len;
+        self.len = i.checked_add(1).expect("Overflow detected");
+        self.set(i, value);
+    }
+
+    /// Removes the last element from the Vob and returns it, or `None` if it is empty.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v = Vob::new();
+    /// v.push(true);
+    /// assert_eq!(v.pop(), Some(true));
+    /// assert_eq!(v.pop(), None);
+    /// ```
+    pub fn pop(&mut self) -> Option<bool> {
+        if self.len == 0 {
+            return None;
+        }
+        // The subtraction can't underflow because self.len > 0.
+        let v = self.get(self.len - 1);
+        debug_assert!(v.is_some());
+        self.len -= 1;
+        self.mask_last_block();
+        v
+    }
+
+    /// Returns the number of elements in the Vob.
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Returns true if the Vob has a length of 0.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// assert_eq!(Vob::from_elem(2, true).is_empty(), false);
+    /// assert_eq!(Vob::new().is_empty(), true);
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Splits the collection into two at the given index.
+    ///
+    /// Returns a newly allocated Self. self contains elements [0, at), and the returned Self
+    /// contains elements [at, len).
+    ///
+    /// Note that the capacity of self does not change.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut v1 = Vob::new();
+    /// v1.push(true);
+    /// v1.push(false);
+    /// let v2 = v1.split_off(1);
+    /// assert_eq!(v1, Vob::from_elem(1, true));
+    /// assert_eq!(v2, Vob::from_elem(1, false));
+    /// ```
+    pub fn split_off(&mut self, at: usize) -> Vob<T> {
+        if at >= self.len {
+            return Vob::<T>::new_with_storage_type(0);
+        }
+        let mut nv = Vob::<T>::new_with_storage_type(self.len - at);
+        // This could easily be made more efficient.
```

Comment:
Indeed. Obvious first step is to copy all "complete" storage blocks using a slice clone, and special case the other remaining block, if there is one. Can you raise an issue?

Reply:
I'd rather wait until someone complains (does anyone use `split_off`? I never have...). It's only inefficient if someone has a very particular use case, at which point we can actually benchmark things.

Reply:
Fine.